csrc/executor.h

```diff
@@ -164,11 +170,26 @@ class FusionExecutor : public NonCopyable {
     post_lowering_hooks_.push_back(std::move(hook));
   }
 
+  // Function to query whether compilation was attempted for a `FusionExecutor`
+  bool isCompiled() const {
+    // Check atmost one of fusion_ and lowered_ is null.
+    NVF_ERROR(
+        fusion_ == nullptr || !lowered_,
+        "Expected GPU lowering to be skipped.");
+    NVF_ERROR(
+        lowered_ == nullptr || !fusion_,
+        "fusion_ should only be initializaed when using expression evaluator.");
```

Comment:
out of curiosity, `!x` is just to check `x != nullptr`, right?

I don't see how the above check is associated with the error message. We are only checking that there's one initialized at most, so that's a `!(fusion_ && lowered_)`